build: cannot compile on OpenBSD since libsecp256k1 subtree update

Since the secp256k1 subtree update in #19228 (8c97780db8c9dd33efed134385573ba97e9cd165), it hasn't been possible to compile on OpenBSD. 
Building fails with (cc is Clang 8.0.1):
```bash
./autogen.sh
./configure --disable-wallet --with-gui=no CC=cc CXX=c++ MAKE=gmake
gmake -j6 V=1
...
gmake[3]: Entering directory '/home/vagrant/bitcoin/src/secp256k1'
gcc -I. -I./src -Wall -Wextra -Wno-unused-function -g -O2 -c src/gen_context.c -o gen_context.o
In file included from src/gen_context.c:16:
src/util.h:179: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'uint128_t'
gmake[3]: *** [Makefile:1689: gen_context.o] Error 1
gmake[3]: Leaving directory '/home/vagrant/bitcoin/src/secp256k1'
```

The issue is that libsecp does some native compiler detection in configure, and then uses that compiler specifically for the ecmult precomputation. See here, (the call settings `CC_FOR_BUILD` is earlier):
https://github.com/bitcoin/bitcoin/blob/476436b2dec254bb988f8c7a6cbec1d7bb7cecfd/src/secp256k1/configure.ac#L186-L188
and then here where it's used:
 https://github.com/bitcoin/bitcoin/blob/476436b2dec254bb988f8c7a6cbec1d7bb7cecfd/src/secp256k1/Makefile.am#L129-L130

The problem is that this ends up picking up and using OpenBSDs ancient GCC 4.2.1, which barfs when it gets to util.h:
https://github.com/bitcoin/bitcoin/blob/476436b2dec254bb988f8c7a6cbec1d7bb7cecfd/src/secp256k1/src/util.h#L179

If anyone does happen to run into this issue, the immediate workaround is to just pass `CC_FOR_BUILD` to configure, i.e: `./configure --disable-wallet --with-gui=no CC=cc CXX=c++ MAKE=gmake CC_FOR_BUILD=cc`.

However before we update any docs etc, I'm going to take the issue upstream.

--- a/doc/build-openbsd.md
+++ b/doc/build-openbsd.md
@@ -1,6 +1,6 @@
 OpenBSD build guide
 ======================
-(updated for OpenBSD 6.4)
+(updated for OpenBSD 6.7)
 
 This guide describes how to build bitcoind and command-line utilities on OpenBSD.
 
@@ -15,7 +15,7 @@ Run the following as root to install the base dependencies for building:
 pkg_add git gmake libevent libtool boost
 pkg_add autoconf # (select highest version, e.g. 2.69)
 pkg_add automake # (select highest version, e.g. 1.16)
-pkg_add python # (select highest version, e.g. 3.6)
+pkg_add python # (select highest version, e.g. 3.8)
 
 git clone https://github.com/bitcoin/bitcoin.git
 ```
@@ -23,10 +23,10 @@ git clone https://github.com/bitcoin/bitcoin.git
 See [dependencies.md](dependencies.md) for a complete overview.
 
 **Important**: From OpenBSD 6.2 onwards a C++11-supporting clang compiler is
-part of the base image, and while building it is necessary to make sure that this
-compiler is used and not ancient g++ 4.2.1. This is done by appending
-`CC=cc CXX=c++` to configuration commands. Mixing different compilers
-within the same executable will result in linker errors.
+part of the base image, and while building it is necessary to make sure that
+this compiler is used and not ancient g++ 4.2.1. This is done by appending
+`CC=cc CC_FOR_BUILD=cc CXX=c++` to configuration commands. Mixing different
+compilers within the same executable will result in errors.
 
 ### Building BerkeleyDB
 
@@ -77,7 +77,7 @@ To configure with wallet:
 
 To configure without wallet:
 ```bash
-./configure --disable-wallet --with-gui=no CC=cc CXX=c++ MAKE=gmake
+./configure --disable-wallet --with-gui=no CC=cc CC_FOR_BUILD=cc CXX=c++ MAKE=gmake
 ```
 
 Build and run the tests:

